src/App.tsx

```diff
@@ -1,24 +1,158 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Error } from './Error';
+import { Header } from './Header';
+import { Todo } from './types/Todo';
+import { TodoList } from './TodoList';
+import { Footer } from './Footer';
+import * as postService from './api/todos';
+import { Status } from './types/Status';
 
-const USER_ID = 0;
+export const USER_ID = 11894;
+
+function getVisibleTodos(todos: Todo[], newStatus: Status) {
+  switch (newStatus) {
+    case Status.ACTIVE:
+      return todos.filter(todo => !todo.completed);
+
+    case Status.COMPLETED:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
```

Comment:
looks like we can move this state to Header, and decrease the number of App re-renders